api/api.go

```diff
@@ -340,13 +342,30 @@ func (a *API) submitBlock(w http.ResponseWriter, r *http.Request) {
 
 	w.Header().Set("Content-Type", "application/json")
 	uc := structs.UserContent{IP: r.Header.Get("X-Forwarded-For")}
-	var l = a.l.With(log.F{"ip": uc.IP})
+	
+	isGzip := strings.ToLower(r.Header.Get("Content-Encoding")) == "gzip"
+	var l = a.l.With(log.F{"ip": uc.IP, "gzip": isGzip})
```

Comment:
Wouldn't it be better to have "encode" or "compression", and "gzip"/"none" as a type? :)

Reply:
a `var` for every encode type you mean?

Reply:
To have information about encoding in logs rather than just "if it's gzip"